# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1530305281 14400
#      Fri Jun 29 16:48:01 2018 -0400
# Node ID d7aafcb1dc1ab97c4d45ff5b40da7c30fd42f082
# Parent  48a9bd2d0a20f285a6054dfa10b7c63171aeb9bb
# Parent  c05adc188b985f46a43ab3ab8e0abea81d75d09a
maint: Merge stable to default.

diff --git a/libgui/src/m-editor/octave-qscintilla.cc b/libgui/src/m-editor/octave-qscintilla.cc
--- a/libgui/src/m-editor/octave-qscintilla.cc
+++ b/libgui/src/m-editor/octave-qscintilla.cc
@@ -112,16 +112,19 @@ namespace octave
              this, SLOT (text_changed (void)));
 
     connect (this, SIGNAL (cursorPositionChanged (int, int)),
              this, SLOT (cursor_position_changed (int, int)));
 
     // clear scintilla edit shortcuts that are handled by the editor
     QsciCommandSet *cmd_set = standardCommands ();
 
+    // Disable buffered drawing on all systems
+    SendScintilla (SCI_SETBUFFEREDDRAW, false);
+
 #if defined (HAVE_QSCI_VERSION_2_6_0)
     // find () was added in QScintilla 2.6
     cmd_set->find (QsciCommand::SelectionCopy)->setKey (0);
     cmd_set->find (QsciCommand::SelectionCut)->setKey (0);
     cmd_set->find (QsciCommand::Paste)->setKey (0);
     cmd_set->find (QsciCommand::SelectAll)->setKey (0);
     cmd_set->find (QsciCommand::SelectionDuplicate)->setKey (0);
     cmd_set->find (QsciCommand::LineTranspose)->setKey (0);
diff --git a/liboctave/external/blas-xtra/xerbla.c b/liboctave/external/blas-xtra/xerbla.c
--- a/liboctave/external/blas-xtra/xerbla.c
+++ b/liboctave/external/blas-xtra/xerbla.c
@@ -54,18 +54,18 @@ octave_set_xerbla_handler (xerbla_handle
    optionally installed handler function.  */
 
 F77_RET_T
 F77_FUNC (xerbla, XERBLA) (F77_CONST_CHAR_ARG_DEF (s_arg, len),
                            const F77_INT *info
                            F77_CHAR_ARG_LEN_DEF (len))
 {
   const char *s = F77_CHAR_ARG_USE (s_arg);
-  size_t slen = F77_CHAR_ARG_LEN_USE (s_arg, len);
+  int slen = F77_CHAR_ARG_LEN_USE (s_arg, len);
 
-  fprintf (stderr, "%.*s: parameter number %ld is invalid\n", slen, s,
+  fprintf (stderr, "%.*s: parameter number %d is invalid\n", slen, s,
            (long) (*info));
 
    if (xerbla_handler)
      (*xerbla_handler) ();
 
   F77_RETURN (0)
 }
